Can't attach to processes using the muxer

When using `PULUMI_DEBUG_PROVIDERS` with a bridged provider using the muxer, it panics on attach.

The `host` field on `m` is nil in the following method. I would assume this is due to the host not being defined if we've started the provider manually instead of via the engine.

I would assume that this is likely fixed by only calling `Close` if `m.host` is not nil.

Relevant code:
```golang
func (m *muxer) Attach(ctx context.Context, req *rpc.PluginAttach) (*emptypb.Empty, error) {
	host, err := provider.NewHostClient(req.GetAddress())
	if err != nil {
		return nil, err
	}
	if err := m.host.Close(); err != nil {
		return nil, err
	}
	// Here we override the underlying host. This should replace the host instance of
	// each subsidiary provider.
	*m.host = *host
	return &emptypb.Empty{}, nil
}
```

Stacktrace from attaching to the AWS provider:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10a329460]

goroutine 41 [running]:
github.com/pulumi/pulumi/pkg/v3/resource/provider.(*HostClient).Close(...)
        /Users/dan/go/pkg/mod/github.com/pulumi/pulumi/pkg/v3@v3.75.0/resource/provider/host.go:62
github.com/pulumi/pulumi-terraform-bridge/x/muxer.(*muxer).Attach(0x14000cb94f0, {0x10e3f58cc?, 0x40?}, 0x11054d300?)
        /Users/dan/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/x/muxer@v0.0.4/muxer.go:401 +0x50
github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Attach_Handler.func1({0x113173b68, 0x14000774c00}, {0x111f90da0?, 0x1400077c240})
        /Users/dan/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.75.0/proto/go/provider_grpc.pb.go:699 +0x74
github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1({0x113173b68, 0x140007749c0}, {0x111f90da0, 0x1400077c240}, 0x1400205a180, 0x14000764330)
        /Users/dan/go/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x308
github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Attach_Handler({0x112e0d540?, 0x14000cb94f0}, {0x113173b68, 0x140007749c0}, 0x14000fba150, 0x14002e34000)
        /Users/dan/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.75.0/proto/go/provider_grpc.pb.go:701 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400034e3c0, {0x113183880, 0x14003a6e1a0}, 0x14005adc120, 0x14005a5aae0, 0x11c7e6b48, 0x0)
        /Users/dan/go/pkg/mod/google.golang.org/grpc@v1.56.1/server.go:1337 +0xc90
google.golang.org/grpc.(*Server).handleStream(0x1400034e3c0, {0x113183880, 0x14003a6e1a0}, 0x14005adc120, 0x0)
        /Users/dan/go/pkg/mod/google.golang.org/grpc@v1.56.1/server.go:1714 +0x82c
google.golang.org/grpc.(*Server).serveStreams.func1.1()
        /Users/dan/go/pkg/mod/google.golang.org/grpc@v1.56.1/server.go:959 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func1
        /Users/dan/go/pkg/mod/google.golang.org/grpc@v1.56.1/server.go:957 +0x16c
```

--- a/x/muxer/muxer.go
+++ b/x/muxer/muxer.go
@@ -398,8 +398,10 @@ func (m *muxer) Attach(ctx context.Context, req *rpc.PluginAttach) (*emptypb.Emp
 	if err != nil {
 		return nil, err
 	}
-	if err := m.host.Close(); err != nil {
-		return nil, err
+	if m.host != nil {
+		if err := m.host.Close(); err != nil {
+			return nil, err
+		}
 	}
 	// Here we override the underlying host. This should replace the host instance of
 	// each subsidiary provider.

